### PR TITLE
Optable 370 ter

### PR DESCRIPTION
**Pull request by abekornelis**
Make optable(370) compatible with HLASM
Move mnemonic ACALLPRM into separate optable
Change default optable from UNI to DFLT - which is UNI plus the ACALLPRM mnemonic
Fix various batch scripts used for building/testing z390
Fix a few documentation shortcomings

### CODE REVIEW

File: bat/BLDJAR.BAT [hunk 1]
```diff
@@ -32,7 +35,8 @@ copy  src\z390.man jar
 cd jar
 rem compile z390 java sources to class vm object code
 java -version
-javac.exe -g:none *.java
+if /I %1. EQU .Debug. javac.exe -g:none *.java
+if /I %1. NEQ .Debug. javac.exe -g      *.java
```
Comment (jyganci) [inline bat/BLDJAR.BAT]: EQU and NEQ should be exchanged. See https://docs.oracle.com/en/java/javase/18/docs/specs/man/javac.html.

Recommend using if else instead of 2 if statements.

File: guam/demo/DEMOGUI1.BAT [hunk 2]
```diff
@@ -1 +1 @@
-asmlg demo\demogui1 gui test trace
+bat\asmlg guam\demo\demogui1 guam test trace
```
Comment (jyganci) [inline guam/demo/DEMOGUI1.BAT:1]: If you are positioned in the z390 directory, then this is correct. Is the invocation of DEMOGUI1.BAT done then?

Just asking.
```

File: doc/user_guide/z390/z390_options.md [hunk 3]
```diff
@@ -70,67 +71,110 @@ MAXSYM(50000) | MA | 50000 | Maximum symbols in one macro assembly.
...
 WRITENONPRINTABLE | MA | NO | Write non-printable characters to PRN, BAL and PCH files. The default value NOWRITENONPRINTABLE replaces all non-printable characters with the '.' character. This avoids issues caused by writing control characters such as 0x0A (Line Feed) or 0x0D (Carriage Return).
 XREF | A | YES | Cross reference symbols to source lines in PRN assembly listing.
-ZSTRMAC | M | YES | Support expansion of ZSTRMAC structured conditional macro code instructions during loading of MLC, MAC, and CPY files.   Note z390 macros in z390\mac directory now are using zstrmac for structured coding. NOZSTRMAC can only be used with MVS, VSE, or other non-structured macro libraries.
+ZSTRMAC | M | YES | Support expansion of ZSTRMAC structured conditional macro code instructions during loading of MLC, MAC, and CPY files. Note z390 macros in z390\mac directory now are using zstrmac for structured coding. NOZSTRMAC can only be used with MVS, VSE, or other non-structured macro libraries.
 ZVSAM | E | 0 | ZVSAM 0=no vsam, 1=zvsam1 support, 2=zvsam2 support.
 
+# Machine and optable options and equivalences
```
Comment (adelosa) [inline doc/user_guide/z390/z390_options.md:139]: Replace with the following text.. The below works with some/most markdown parsers but something is upsetting the GitHub one.

```text
| Machine    | Machine   | Optable | Optable                                                                     | Description      |
|------------|-----------|---------|-----------------------------------------------------------------------------|------------------|
|            | DOS       |         |                                                                             |                  |
| S370       |           | 370     |                                                                             |                  |
| S370XA     | ARCH-0    | XA      |                                                                             |                  |
| S370ESA    | ARCH-1    | ESA     |                                                                             | Synonyms for ESA |
| S390       | ARCH-2    | ESA     |                                                                             |                  |
| S390E      | ARCH-3    | ESA     |                                                                             |                  |
| ARCH-4     | ESA       |         |                                                                             |                  |
| zSeries    | ZSeries-1 | ZS1     | ZOP                                                                         | Synonyms for ZOP |
| ZS         | ZS-1      | ZS1     | ZOP                                                                         |                  |
| z800       | z900      | ZS1     | ZOP                                                                         |                  |
| ARCH-5     | ZS1       | ZOP     |                                                                             |                  |
| z890       | z990      | ZS2     | YOP                                                                         | Synonyms for YOP |
| zSeries-2  | ZS-2      | ZS2     | YOP                                                                         |                  |
| ARCH-6     | ZS2       | YOP     |                                                                             |                  |
| zSeries-3  | ZS-3      | ZS3     | Z9                                                                          | Synonyms for Z9  |
| z9         | ARCH-7    | ZS3     | Z9                                                                          |                  |
| zSeries-4  | ZS-4      | ZS4     | Z10                                                                         | Synonyms for Z10 |
| z10        | ARCH-8    | ZS4     | Z10                                                                         |                  |
| zSeries-5  | ZS-5      | ZS5     | Z11                                                                         | Synonyms for Z11 |
| z114       | z196      | ZS5     | Z11                                                                         |                  |
| z11        | ARCH-9    | ZS5     | Z11                                                                         |                  |
| zSeries-6  | ZS-6      | ZS6     | Z12                                                                         | Synonyms for Z12 |
| zBC12      | zEC12     | ZS6     | Z12                                                                         |                  |
| z12        | ARCH-10   | ZS6     | Z12                                                                         |                  |
| zSeries-7  | ZS-7      | ZS7     | Z13                                                                         | Synonyms for Z13 |
| z13        | ARCH-11   | ZS7     | Z13                                                                         |                  |
| zSeries-8  | ZS-8      | ZS8     | Z14                                                                         | Synonyms for Z14 |
| z14        | ARCH-12   | ZS8     | Z14                                                                         |                  |
| zSeries-9  | ZS-9      | ZS9     | Z15                                                                         | Synonyms for Z15 |
| z15        | ARCH-13   | ZS9     | Z15                                                                         |                  |
| zSeries-10 | ZS-10     | ZSA     | Z16                                                                         | Synonyms for Z16 |
| z16        | ARCH-14   | ZSA     | Z16                                                                         |                  |
|            | UNI       |         | All current instructions from optables above                                |                  |
|            | [DFLT]    |         | Cannot be specified, same as UNI plus one z390 directive for zCobol support |                  |
|            | z390      |         | UNI plus all z390 instructions; not compatible with HLASM                   |                  |
```

File: guam/demo/DEMOGUI1.BAT (hunk 2 above)
Comment (abekornelis) [reply guam/demo/DEMOGUI1.BAT:1]: I think that for all bat files we have been assuming they'd be invoked from the z390 root dir.
I have applied the same assumption to this script. Would it be more robust and flexible if use something like ~dps0~
(Sorry, I do not have exact syntax available right now. I hope you get the intention)

Comment (jyganci) [reply guam/demo/DEMOGUI1.BAT:1]: The only references to DEMOGUI1 I found are in rt/RTGUAM.BAT, guam/demo/DEMOGUI1.MLC and guam/demo/DEMOGUI2.MLC. There is no invocation of rt/DEMOGUI1.BAT??? Why are you even looking at it?

My guess is that rt/DEMOGUI1.BAT is a standalone bat file meant to be invoked by the user. Since the file is in the same directory as DEMOGUI1.MLC, I assume that the current directory for the user would be guam/demo. Therefore, the current bat file would fail for 2 reasons: 1.  No asmlg in current directory; 2. No demo subidrectory of current directory.

In short, the bat file needs work. I suspect that it was created as a test only.

The rt/RTGUAM.BAT file invokes rt/RT1 for DEMOGUI1 via "call rt\RT1 guam\demo\DEMOGUI1  guam %1 %2 %3 %4 %5 %6 %7 %8 %9", which in turn issues "call ASMLG %1 @rt\RT1 %2 %3 %4 %5 %6 %7 %8 %9". Neither rt/RTGUAM.BAT nor rt/RT1.BAT has been vetted for the current z390.

The GUAM tests are currently not part of our regression tests. To my knowledge, few of the bat files in the rt directory and subdirectories have been looked at for the current z390.

File: guam/test/TEST.BAT [hunk 4]
```diff
@@ -1,5 +1,5 @@
 rem run the tn3270 test tutorial pgms
-call asmlg tn3270\hello1 guam
-call asmlg tn3270\hello2 guam
-call asmlg tn3270\hello3 guam
-call asmlg tn3270\aid guam
+call bat\asmlg guam\demo\hello1 guam
+call bat\asmlg guam\demo\hello2 guam
+call bat\asmlg guam\demo\hello3 guam
+call bat\asmlg guam\test\aid guam
```
Comment (jyganci) [inline guam/test/TEST.BAT]: Same comments as for guam/demo/DEMOGUI1.BAT.

If you decide to go ahead with this cnage, then change filenames to uppercase: HELLO1, HELLO2, HELLO3, AID.

File: rt/rt/OPCD$.MLC [hunk 5]
```diff
@@ -1,13 +1,54 @@
+OPCD$DOS  TITLE 'Validate opcode table'
```
Comment (jyganci) [inline rt/rt/OPCD$.MLC]: Should "OPCD$DOS" be changed to "OPCD$"? If yes, same change on the CSECT and USING lines below.

File: rt/rt/OPCD$.MLC [hunk 6]
```diff
@@ -1,13 +1,54 @@
...
+         YREGS ,                       * register equates
+
+***********************************************************************
+*
+* Program entry and linkage
+*
+***********************************************************************
+OPCD$DOS CSECT ,                       *
+*PCD$DOS AMODE 24                      * Not supported in DOS
+*PCD$DOS RMODE 24                      * Not supported in DOS
```
Comment (jyganci) [inline rt/rt/OPCD$.MLC]: Delete these 2 lines???

File: guam/demo/DEMOGUI1.BAT (hunk 2 above)
Comment (abekornelis) [reply guam/demo/DEMOGUI1.BAT:1]: I think I need to run all tests, not just the ones that were included in the automated test suite. Compiling zCics has always been a major touchstone for z390. It is okay not to have these tests run every push, but I do want to be able to run these tests when needed. So yes, there is a procedure that calls it, but it is not part of z390. For now it lives outside z390 on my laptop. I have at times considered proposing a change to the build procedure to add a *ALL parameter, but have not yet done that.

File: rt/rt/OPCD$.MLC (hunk 6 above)
Comment (abekornelis) [reply rt/rt/OPCD$.MLC]: These two statements do not work when assembling with optable(DOS).
I encapsulated them under an AIF that tests whether these mnemonics are defined.

File: src/tz390.java [hunk 7]
```diff
@@ -1824,6 +1810,10 @@ public  class  tz390 {
         {"0D=BASR,RR,20",        //    320 "0D"    "BASR"     "RR"    2 // RPI 1209N
          "4D=BAS,5,50",          //   1150 "4D"    "BAS"      "RX"    5
          "AE=SIGP,10,100",       //   2540 "AE"    "SIGP"     "RS"   10
+         "9C02=RIO,7,70",        //        "9C02"  "RIO"      "S"     7 // #533
+         "9F01=CLRCH,7,70",      //        "9F01"  "CLRCH"    "S"     7 // #533
```
Comment (jyganci) [inline src/tz390.java:1813]: Should these 2 lines be above the preceding line??? Maybe 2 byte instructions follow 1 byte instructions???

File: rt/rt/OPCD$.MLC (hunk 5 above)
Comment (abekornelis) [reply rt/rt/OPCD$.MLC]: Correct, done.

File: src/tz390.java (hunk 7 above)
Comment (abekornelis) [reply src/tz390.java:1813]: yes, that would be better. Thanks!

**Review by jyganci — APPROVED** [review]
